File: source/bxgenbb_help/src/decay0_fortran/funbeta.f
c funbeta.f 
c
c This file was extracted from the 'decay0' program by V.I. Tretyak
c
c Copyright 1995-2011 V.I. Tretyak
c
c This program is free software; you can redistribute it and/or modify
c it under the terms of the GNU General Public License as published by
c the Free Software Foundation; either version 3 of the License, or (at
c your option) any later version.
c 
c This program is distributed in the hope that it will be useful, but
c WITHOUT ANY WARRANTY; without even the implied warranty of
c MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
c General Public License for more details.
c 
c You should have received a copy of the GNU General Public License
c along with this program; if not, write to the Free Software
c Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
c 
c
	function funbeta(E)
c Function to search the maximum of beta spectrum curve by tgold
c subroutine.
c VIT, 30.07.1992; 15.10.1995.
	common/const/pi,emass,datamass(50)
	common/parbeta/Zdtr,Qbeta
	funbeta=0.
	if(E.gt.0.) funbeta=sqrt(E*(E+2.*emass))*(E+emass)*(Qbeta-E)**2
     +                   *fermi(Zdtr,E)
	return
	end
c
c end of funbeta.f
c
c Local Variables: --
c mode: fortran --
c End: --

File: source/bxgenbb_help/src/decay0/funbeta.cc
// -*- mode: c++; -*- 
// fermi.cc

#include <iostream>
#include <cmath>
#include <stdexcept>

#include <genbb_help/decay0/funbeta.h>
#include <genbb_help/decay0/fermi.h>
#include <genbb_help/decay0/particle.h>

#include <gsl/gsl_math.h>

namespace genbb {
  namespace decay0 {
   
    /// \brief Function to search the maximum of beta spectrum curve by decay0_tgold subroutine.
    double decay0_funbeta(double E_, void * params_)
    {
      const parbeta * pars = static_cast<const parbeta *>(params_);
      const double & Qbeta = pars->Qbeta;
      const double & Zdtr = pars->Zdtr;
      double f = 0.0;
      static const double emass = decay0_emass();
      if (E_ > 0.) {
        f = std::sqrt(E_ * (E_ + 2. * emass)) 
          * (E_ + emass) 
          * gsl_pow_2(Qbeta - E_)
          * decay0_fermi(Zdtr, E_);
      }
      return f;
    }

  } // end of namespace decay0 
} // end of namespace genbb 
